components/org.wso2.carbon.identity.conditional.auth.functions.elk/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/elk/CallElasticFunctionImpl.java

```diff
@@ -0,0 +1,200 @@
+
+                request.setHeader(ACCEPT, TYPE_APPLICATION_JSON);
+                request.setHeader(CONTENT_TYPE, TYPE_APPLICATION_JSON);
+                handleAuthentication(request, authenticationContext.getTenantDomain());
+
+                String query = elasticConfigProvider.getQuery(params);
+                request.setEntity(new StringEntity(query, StandardCharsets.UTF_8));
+
+                String[] targetHostUrls = targetHostUrl.split(";");
+
+                HttpHost[] targetHosts = new HttpHost[targetHostUrls.length];
+
+                for (int i = 0; i < targetHostUrls.length; i++) {
+                    URL hostUrl = new URL(targetHostUrls[i]);
+                    targetHosts[i] = new HttpHost(hostUrl.getHost(), hostUrl.getPort(), hostUrl.getProtocol());
+                }
+
+                CloseableHttpAsyncClient client = HttpClientManager.getInstance().getClient(tenantDomain);
+
+                AtomicInteger requestAtomicInteger = new AtomicInteger(targetHosts.length);
+
+                for (final HttpHost targetHost : targetHosts) {
+                    client.execute(targetHost, request, new FutureCallback<HttpResponse>() {
+
+                        @Override
+                        public void completed(final HttpResponse response) {
+
+                            int responseCode = response.getStatusLine().getStatusCode();
+                            try {
+                                Map<String, Object> responseMap = new HashMap<>();
+                                LOG.error(responseCode);
+                                if (responseCode == 200) {
+                                    try {
+                                        String jsonString = EntityUtils.toString(response.getEntity());
+                                        JSONObject responseBody = new JSONObject(jsonString);
+                                        int score = responseBody
+                                                .getJSONObject("aggregations")
+                                                .getJSONObject("risk_score")
+                                                .getInt("value");
+                                        responseMap.put("risk_score", score);
+                                        asyncReturn.accept(authenticationContext, responseMap, OUTCOME_SUCCESS);
+                                    } catch (JSONException e) {
+                                        LOG.error("Error while building response from analytics engine call for " +
+                                                "session data key: " + authenticationContext.getContextIdentifier(), e);
+                                        asyncReturn.accept(authenticationContext, responseMap, OUTCOME_FAIL);
+                                    } catch (IOException e) {
+                                        LOG.error("Error while reading response from analytics engine call for " +
```

Comment:
Wouldn't it be better if we print the exception `e` too in the error log?

Reply:
Resolved in https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/110